.../storage/connect/plgxml.h:10:32: error: 'XML_ELEMENT_NODE' conflicts with a previous declaration
   10 |      XML_ELEMENT_NODE       =  1,
      |                                ^
.../opt/include/libxml2/libxml/tree.h:175:5: note: previous declaration 'xmlElementType XML_ELEMENT_NODE'
  175 |     XML_ELEMENT_NODE=           1,
      |     ^~~~~~~~~~~~~~~~
.../storage/connect/plgxml.h:11:32: error: 'XML_ATTRIBUTE_NODE' conflicts with a previous declaration
   11 |      XML_ATTRIBUTE_NODE     =  2,
      |                                ^
.../opt/include/libxml2/libxml/tree.h:176:5: note: previous declaration 'xmlElementType XML_ATTRIBUTE_NODE'
  176 |     XML_ATTRIBUTE_NODE=         2,
      |     ^~~~~~~~~~~~~~~~~~
.../storage/connect/plgxml.h:12:32: error: 'XML_TEXT_NODE' conflicts with a previous declaration
   12 |      XML_TEXT_NODE          =  3,
      |                                ^
.../opt/include/libxml2/libxml/tree.h:177:5: note: previous declaration 'xmlElementType XML_TEXT_NODE'
  177 |     XML_TEXT_NODE=              3,
      |     ^~~~~~~~~~~~~
.../storage/connect/plgxml.h:13:32: error: 'XML_CDATA_SECTION_NODE' conflicts with a previous declaration
   13 |      XML_CDATA_SECTION_NODE =  4,
      |                                ^
.../opt/include/libxml2/libxml/tree.h:178:5: note: previous declaration 'xmlElementType XML_CDATA_SECTION_NODE'
  178 |     XML_CDATA_SECTION_NODE=     4,
      |     ^~~~~~~~~~~~~~~~~~~~~~
.../storage/connect/plgxml.h:14:32: error: 'XML_ENTITY_REF_NODE' conflicts with a previous declaration
   14 |      XML_ENTITY_REF_NODE    =  5,
      |                                ^
.../opt/include/libxml2/libxml/tree.h:179:5: note: previous declaration 'xmlElementType XML_ENTITY_REF_NODE'
  179 |     XML_ENTITY_REF_NODE=        5,
      |     ^~~~~~~~~~~~~~~~~~~
.../storage/connect/plgxml.h:15:32: error: 'XML_ENTITY_NODE' conflicts with a previous declaration
   15 |      XML_ENTITY_NODE        =  6,
      |                                ^
.../opt/include/libxml2/libxml/tree.h:180:5: note: previous declaration 'xmlElementType XML_ENTITY_NODE'
  180 |     XML_ENTITY_NODE=            6,  /* unused */
      |     ^~~~~~~~~~~~~~~
.../storage/connect/plgxml.h:16:32: error: 'XML_PI_NODE' conflicts with a previous declaration
   16 |      XML_PI_NODE            =  7,
      |                                ^
.../opt/include/libxml2/libxml/tree.h:181:5: note: previous declaration 'xmlElementType XML_PI_NODE'
  181 |     XML_PI_NODE=                7,
      |     ^~~~~~~~~~~
.../storage/connect/plgxml.h:17:32: error: 'XML_COMMENT_NODE' conflicts with a previous declaration
   17 |      XML_COMMENT_NODE       =  8,
      |                                ^
.../opt/include/libxml2/libxml/tree.h:182:5: note: previous declaration 'xmlElementType XML_COMMENT_NODE'
  182 |     XML_COMMENT_NODE=           8,
      |     ^~~~~~~~~~~~~~~~
.../storage/connect/plgxml.h:18:32: error: 'XML_DOCUMENT_NODE' conflicts with a previous declaration
   18 |      XML_DOCUMENT_NODE      =  9,
      |                                ^
.../opt/include/libxml2/libxml/tree.h:183:5: note: previous declaration 'xmlElementType XML_DOCUMENT_NODE'
  183 |     XML_DOCUMENT_NODE=          9,
      |     ^~~~~~~~~~~~~~~~~
.../storage/connect/plgxml.h:19:31: error: 'XML_DOCUMENT_TYPE_NODE' conflicts with a previous declaration
   19 |      XML_DOCUMENT_TYPE_NODE = 10,
      |                               ^~
.../opt/include/libxml2/libxml/tree.h:184:5: note: previous declaration 'xmlElementType XML_DOCUMENT_TYPE_NODE'
  184 |     XML_DOCUMENT_TYPE_NODE=     10, /* unused */
      |     ^~~~~~~~~~~~~~~~~~~~~~
.../storage/connect/plgxml.h:20:31: error: 'XML_DOCUMENT_FRAG_NODE' conflicts with a previous declaration
   20 |      XML_DOCUMENT_FRAG_NODE = 11,
      |                               ^~
.../opt/include/libxml2/libxml/tree.h:185:5: note: previous declaration 'xmlElementType XML_DOCUMENT_FRAG_NODE'
  185 |     XML_DOCUMENT_FRAG_NODE=     11,
      |     ^~~~~~~~~~~~~~~~~~~~~~
.../storage/connect/plgxml.h:21:31: error: 'XML_NOTATION_NODE' conflicts with a previous declaration
   21 |      XML_NOTATION_NODE      = 12,
      |                               ^~
.../opt/include/libxml2/libxml/tree.h:186:5: note: previous declaration 'xmlElementType XML_NOTATION_NODE'
  186 |     XML_NOTATION_NODE=          12, /* unused */
      |     ^~~~~~~~~~~~~~~~~
.../storage/connect/plgxml.h:22:31: error: 'XML_HTML_DOCUMENT_NODE' conflicts with a previous declaration
   22 |      XML_HTML_DOCUMENT_NODE = 13,
      |                               ^~
.../opt/include/libxml2/libxml/tree.h:187:5: note: previous declaration 'xmlElementType XML_HTML_DOCUMENT_NODE'
  187 |     XML_HTML_DOCUMENT_NODE=     13,
      |     ^~~~~~~~~~~~~~~~~~~~~~
.../storage/connect/plgxml.h:23:31: error: 'XML_DTD_NODE' conflicts with a previous declaration
   23 |      XML_DTD_NODE           = 14,
      |                               ^~
.../opt/include/libxml2/libxml/tree.h:188:5: note: previous declaration 'xmlElementType XML_DTD_NODE'
  188 |     XML_DTD_NODE=               14,
      |     ^~~~~~~~~~~~
.../storage/connect/plgxml.h:24:31: error: 'XML_ELEMENT_DECL' conflicts with a previous declaration
   24 |      XML_ELEMENT_DECL       = 15,
      |                               ^~
.../opt/include/libxml2/libxml/tree.h:189:5: note: previous declaration 'xmlElementType XML_ELEMENT_DECL'
  189 |     XML_ELEMENT_DECL=           15,
      |     ^~~~~~~~~~~~~~~~
.../storage/connect/plgxml.h:25:31: error: 'XML_ATTRIBUTE_DECL' conflicts with a previous declaration
   25 |      XML_ATTRIBUTE_DECL     = 16,
      |                               ^~
.../opt/include/libxml2/libxml/tree.h:190:5: note: previous declaration 'xmlElementType XML_ATTRIBUTE_DECL'
  190 |     XML_ATTRIBUTE_DECL=         16,
      |     ^~~~~~~~~~~~~~~~~~
.../storage/connect/plgxml.h:26:31: error: 'XML_ENTITY_DECL' conflicts with a previous declaration
   26 |      XML_ENTITY_DECL        = 17,
      |                               ^~
.../opt/include/libxml2/libxml/tree.h:191:5: note: previous declaration 'xmlElementType XML_ENTITY_DECL'
  191 |     XML_ENTITY_DECL=            17,
      |     ^~~~~~~~~~~~~~~
.../storage/connect/plgxml.h:27:31: error: 'XML_NAMESPACE_DECL' conflicts with a previous declaration
   27 |      XML_NAMESPACE_DECL     = 18,
      |                               ^~
.../opt/include/libxml2/libxml/tree.h:192:5: note: previous declaration 'xmlElementType XML_NAMESPACE_DECL'
  192 |     XML_NAMESPACE_DECL=         18,
      |     ^~~~~~~~~~~~~~~~~~
.../storage/connect/plgxml.h:28:31: error: 'XML_XINCLUDE_START' conflicts with a previous declaration
   28 |      XML_XINCLUDE_START     = 19,
      |                               ^~
.../opt/include/libxml2/libxml/tree.h:193:5: note: previous declaration 'xmlElementType XML_XINCLUDE_START'
  193 |     XML_XINCLUDE_START=         19,
      |     ^~~~~~~~~~~~~~~~~~
.../storage/connect/plgxml.h:29:31: error: 'XML_XINCLUDE_END' conflicts with a previous declaration
   29 |      XML_XINCLUDE_END       = 20,
      |                               ^~
.../opt/include/libxml2/libxml/tree.h:194:5: note: previous declaration 'xmlElementType XML_XINCLUDE_END'
  194 |     XML_XINCLUDE_END=           20
      |     ^~~~~~~~~~~~~~~~
.../opt/include/libxml2/libxml/tree.h:200:32: error: expected identifier before numeric constant
  200 | #define XML_DOCB_DOCUMENT_NODE 21
      |                                ^~
.../storage/connect/plgxml.h:30:6: note: in expansion of macro 'XML_DOCB_DOCUMENT_NODE'
   30 |      XML_DOCB_DOCUMENT_NODE = 21};
      |      ^~~~~~~~~~~~~~~~~~~~~~
.../opt/include/libxml2/libxml/tree.h:200:32: error: expected '}' before numeric constant
  200 | #define XML_DOCB_DOCUMENT_NODE 21
      |                                ^~
.../storage/connect/plgxml.h:30:6: note: in expansion of macro 'XML_DOCB_DOCUMENT_NODE'
   30 |      XML_DOCB_DOCUMENT_NODE = 21};
      |      ^~~~~~~~~~~~~~~~~~~~~~
.../storage/connect/plgxml.h:9:18: note: to match this '{'
    9 | enum ElementType {               // libxml2
      |                  ^
.../opt/include/libxml2/libxml/tree.h:200:32: error: expected unqualified-id before numeric constant
  200 | #define XML_DOCB_DOCUMENT_NODE 21
      |                                ^~
.../storage/connect/plgxml.h:30:6: note: in expansion of macro 'XML_DOCB_DOCUMENT_NODE'
   30 |      XML_DOCB_DOCUMENT_NODE = 21};
      |      ^~~~~~~~~~~~~~~~~~~~~~
.../storage/connect/plgxml.h:30:33: error: expected declaration before '}' token
   30 |      XML_DOCB_DOCUMENT_NODE = 21};
      |                                 ^
ninja: build stopped: subcommand failed.

---

--- a/storage/connect/plgxml.h
+++ b/storage/connect/plgxml.h
@@ -5,7 +5,7 @@
 /******************************************************************/
 /*  Dual XML implementation base classes defines.                 */
 /******************************************************************/
-#if !defined(BASE_BUFFER_SIZE)
+#if !defined(__XML_TREE_H__)
 enum ElementType {               // libxml2
      XML_ELEMENT_NODE       =  1,
      XML_ATTRIBUTE_NODE     =  2,
